client.c

```diff
@@ -0,0 +1,76 @@
+#include "client.h"
+
+#include "brute.h"
+#include "common.h"
+#include "single.h"
+
+#include <netinet/in.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+bool
+run_client (task_t *task, config_t *config)
+{
+  int socket_fd = socket (AF_INET, SOCK_STREAM, 0);
+  if (socket_fd == -1)
+    {
+      print_error ("Could not initialize client socket\n");
+      return (false);
+    }
+
+  struct sockaddr_in addr;
+  addr.sin_family = AF_INET;
+  addr.sin_addr.s_addr = INADDR_ANY;
+  addr.sin_port = htons (9000);
+
+  if (connect (socket_fd, (struct sockaddr *)&addr, sizeof (addr)) == -1)
+    {
+      print_error ("Could not connect to server\n");
+      return (false);
+    }
+
+  st_context_t st_context = {
+    .hash = config->hash,
```

Comment:
We need to get hash from server. I.e. first thing that server sends after connect is a hash.